torch/cuda/amp/grad_scaler.py

```diff
@@ -1,117 +1,12 @@
-from __future__ import annotations
-
-import inspect
-import warnings
-from collections import abc, defaultdict
-from enum import Enum
-from typing import Any, cast, Dict, Iterable, List, Optional, overload, Tuple, Union
-
 import torch
-from .common import amp_definitely_not_available
-
 
-__all__ = ["OptState", "GradScaler"]
+__all__ = ["GradScaler"]
```

Comment:
Removing the OptState export from here is BC breaking, are you sure you want to do this?

Reply:
`OptState` is moved from torch/cuda/amp/grad_scaler.py to torch/amp/grad_scaler.py and it will be used by https://github.com/pytorch/pytorch/pull/109993/files#diff-c063bce8f213b0b5e06358b13e6d1adb885a1428ce601cecf16b4c68a0dfe233L7.
Since the common logic has been moved to torch/amp/grad_scaler.py, export `OptState` from torch/amp/grad_scaler.py seems more reasonable.